tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs

```diff
@@ -14,65 +16,74 @@ public V3RelevancyFunctionalTests(CommonFixture fixture, ITestOutputHelper testO
+            yield return new object[] { "newtonsoft.json", new[] { "newtonsoft.json" } };
+            yield return new object[] { "newtonsoft", new[] { "newtonsoft.json" } };
+            yield return new object[] { "json.net", new[] { "json.net", "newtonsoft.json" } };
+            yield return new object[] { "json", new[] { "json", "newtonsoft.json" } };
 
-            Assert.NotEmpty(results);
-            Assert.Contains("log4net", results);
+            yield return new object[] { "tags:\"aws-sdk-v3\"", new[] { "awssdk.core", "awssdk.s3" } };
 
-            // TODO: These should be on the first page!
-            //Assert.Contains("nlog", results);
-            //Assert.Contains("serilog", results);
-            //Assert.Contains("microsoft.extensions.logging, results);
-        }
+            yield return new object[] { "entityframework", new[] { "entityframework" } };
+            yield return new object[] { "entity framework", new[] { "entity", "entityframework" } };
+            yield return new object[] { "EntityFrameworkCore", new[] { "microsoft.entityframeworkcore" } };
+            yield return new object[] { "microsoft.entityframeworkcore", new[] { "microsoft.entityframeworkcore" } };
+            yield return new object[] { "mysql", new[] { "mysql.data" } };
 
-        [RelevancyFact]
-        public async Task EntityFrameworkCore()
-        {
-            var results = await SearchAsync("EntityFrameworkCore");
+            yield return new object[] { "microsoft.aspnetcore.app", new[] { "microsoft.aspnetcore.app" } };
+            yield return new object[] { "microsoft.extensions.logging", new[] { "microsoft.extensions.logging" } };
 
-            Assert.Equal("microsoft.entityframeworkcore", results[0]);
+            yield return new object[] { "xunit", new[] { "xunit" } };
+            yield return new object[] { "nunit", new[] { "nunit" } };
+            yield return new object[] { "dapper", new[] { "dapper" } };
+            yield return new object[] { "log4net", new[] { "log4net" } };
+            yield return new object[] { "automapper", new[] { "automapper" } };
+            yield return new object[] { "csv", new[] { "csv", "csvhelper" } };
+            yield return new object[] { "bootstrap", new[] { "bootstrap" } };
+            yield return new object[] { "moq", new[] { "moq" } };
+            yield return new object[] { "serilog", new[] { "serilog" } };
+            yield return new object[] { "redis", new[] { "redis", "stackexchange.redis", "microsoft.extensions.caching.redis" } };
         }
 
-        [RelevancyFact]
-        public async Task MicrosoftExtensions()
+        [RelevancyTheory]
+        [MemberData(nameof(EnsureTopResultsData))]
+        public async Task EnsureTopResults(string searchTerm, string[] expectedTopResults)
         {
-            var results = await SearchAsync("Microsoft.Extensions");
+            var results = await SearchAsync(searchTerm);
```

Comment:
i think the number of results returned should be clarified here. For example if 100 is returned, this is not a very strong test.

Reply:
Would prefer a parameter to `SearchAsync` but a comment can work too (but may be out of date). Perhaps an assertion too like asserting the count is less than or equal to something.

Reply:
Added a parameter and explicitly chose the number of results in the test